core/src/main/java/org/elasticsearch/index/query/SpanContainingQueryBuilder.java

```diff
@@ -92,9 +84,16 @@ protected Query doToQuery(QueryParseContext parseContext) throws IOException {
 
     @Override
     public QueryValidationException validate() {
-        QueryValidationException validationExceptions = validateInnerQuery(big, null);
-        validationExceptions = validateInnerQuery(little, validationExceptions);
-        return validationExceptions;
+        QueryValidationException validationException = null;
+        if (big == null) {
+            validationException = addValidationError("inner clause [big] cannot be null.", validationException);
+        }
+        if (little == null) {
+            validationException = addValidationError("inner clause [little] cannot be null.", validationException);
+        }
+        validationException = validateInnerQuery(big, validationException);
+        validationException = validateInnerQuery(little, validationException);
```

Comment:
wondering, I see that we have a null check in validateInnerQuery, I guess for queries that are optional. Just an idea, but maybe we could have a validateInnerQuery that throws error if null  and validateOptionalInnerQuery which only validates if not null like now?